test/unittest/tests/test_binary_search_string.cpp

```diff
@@ -0,0 +1,101 @@
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+
+TEST_CASE("Binary Search String: WT_BINARY_SEARCH_STRING", "[search]")
+{
+    bool found;
+
+    SECTION("Key exists in the array")
+    {
+        const char *array[] = {"apple", "banana", "cherry", "date", "elderberry"};
+        const uint32_t n = sizeof(array) / sizeof(array[0]);
+
+        WT_BINARY_SEARCH_STRING("banana", array, n, found);
+        REQUIRE(found == true);
+
+        WT_BINARY_SEARCH_STRING("elderberry", array, n, found);
+        REQUIRE(found == true);
+
+        WT_BINARY_SEARCH_STRING("apple", array, n, found);
+        REQUIRE(found == true);
+    }
+
+    SECTION("Key does not exist in the array")
+    {
+        const char *array[] = {"apple", "banana", "cherry", "date", "elderberry"};
+        const uint32_t n = sizeof(array) / sizeof(array[0]);
+
+        WT_BINARY_SEARCH_STRING("grape", array, n, found);
+        REQUIRE(found == false);
+
+        WT_BINARY_SEARCH_STRING("kiwi", array, n, found);
+        REQUIRE(found == false);
+
+        WT_BINARY_SEARCH_STRING("fig", array, n, found);
+        REQUIRE(found == false);
+    }
+
+    SECTION("Empty array")
+    {
+        const char *array[] = {};
+        const uint32_t n = sizeof(array) / sizeof(array[0]);
+
+        WT_BINARY_SEARCH_STRING("apple", array, n, found);
+        REQUIRE(found == false);
+
+        WT_BINARY_SEARCH_STRING("banana", array, n, found);
+        REQUIRE(found == false);
+
+        WT_BINARY_SEARCH_STRING("cherry", array, n, found);
+        REQUIRE(found == false);
+    }
+
+    SECTION("Single element array")
+    {
+        const char *array[] = {"apple"};
+        const uint32_t n = sizeof(array) / sizeof(array[0]);
+
+        WT_BINARY_SEARCH_STRING("apple", array, n, found);
+        REQUIRE(found == true);
+
+        WT_BINARY_SEARCH_STRING("banana", array, n, found);
+        REQUIRE(found == false);
+    }
+
+    SECTION("Array with duplicate elements")
+    {
+        const char *array[] = {
+          "apple", "banana", "banana", "cherry", "date", "elderberry", "elderberry"};
+        const uint32_t n = sizeof(array) / sizeof(array[0]);
+
+        WT_BINARY_SEARCH_STRING("banana", array, n, found);
+        REQUIRE(found == true);
+
+        WT_BINARY_SEARCH_STRING("elderberry", array, n, found);
+        REQUIRE(found == true);
+
+        WT_BINARY_SEARCH_STRING("fig", array, n, found);
+        REQUIRE(found == false);
+    }
+
+    SECTION("Key exists in the array and is the first or last element")
+    {
+        const char *array[] = {"apple", "banana", "cherry", "date", "elderberry"};
+        const uint32_t n = sizeof(array) / sizeof(array[0]);
+
+        WT_BINARY_SEARCH_STRING("apple", array, n, found);
+        REQUIRE(found == true);
+
+        WT_BINARY_SEARCH_STRING("elderberry", array, n, found);
+        REQUIRE(found == true);
+    }
```

Comment:
These are pretty good - I'd suggest a few more cases, e.g.
* String prefixes matching, e.g. `array = {"apple", ...}` and search for `apples`
* The other way around - have `apples` in the array and search for `apple`
* Empty strings. Is `""` in the normal array of fruit? What about if you have `array = {"", "apple", ...}`?
* Vice versa - if `array = {"", "", ...}`, is the empty string found? What about NULL?

Reply:
Nice one, will add those.

Reply:
@korteland I have added the test scenarios that you requested.

Also, the compiler starts to complain if the key is NULL. Hence, we should be good on that end. Also, binary search is unsupported if there is a NULL value present in an array. I have added a comment. 

The above is updated in [d68675c](https://github.com/wiredtiger/wiredtiger/pull/9438/commits/d68675c7854985f346b86de70f485ec92c4b5589)